### ISSUE TITLE
SNOW-200542: pip install snowflake-connector-python[pandas] raises error on python 3.8.5

### ISSUE THREAD

**Issue by mlduarte** [snowflakedb/snowflake-connector-python#449]
1. What version of Python are you using (`python --version`)? 
>>>Python 3.8.5

2. What operating system and processor architecture are you using (`python -c 'import platform; print(platform.platform())'`)?
>>>Windows-10-10.0.19041-SP0

3. What are the component versions in the environment (`pip freeze`)?

```
argon2-cffi @ file:///C:/ci/argon2-cffi_1596828585465/work
async-generator==1.10
attrs @ file:///tmp/build/80754af9/attrs_1600298409949/work
backcall==0.2.0
bleach @ file:///tmp/build/80754af9/bleach_1600439572647/work
certifi==2020.6.20
cffi @ file:///C:/ci/cffi_1600699246375/work
colorama==0.4.3
decorator==4.4.2
defusedxml==0.6.0
entrypoints==0.3
importlib-metadata @ file:///C:/ci/importlib-metadata_1593446511143/work
ipykernel @ file:///C:/ci/ipykernel_1596190155316/work/dist/ipykernel-5.3.4-py3-none-any.whl
ipython @ file:///C:/ci/ipython_1598865816163/work
ipython-genutils==0.2.0
ipywidgets==7.5.1
jedi @ file:///C:/ci/jedi_1598353661586/work
Jinja2==2.11.2
jsonschema==3.2.0
jupyter==1.0.0
jupyter-client @ file:///tmp/build/80754af9/jupyter_client_1601311786391/work
jupyter-console @ file:///tmp/build/80754af9/jupyter_console_1598884538475/work
jupyter-core==4.6.3
jupyterlab-pygments @ file:///tmp/build/80754af9/jupyterlab_pygments_1600976720582/work
MarkupSafe==1.1.1
mistune==0.8.4
nbclient @ file:///tmp/build/80754af9/nbclient_1601059699549/work
nbconvert @ file:///C:/ci/nbconvert_1601051200066/work
nbformat==5.0.7
nest-asyncio @ file:///tmp/build/80754af9/nest-asyncio_1600977170767/work
notebook @ file:///C:/ci/notebook_1596840175976/work
packaging==20.4
pandocfilters==1.4.2
parso==0.7.0
pickleshare==0.7.5
prometheus-client==0.8.0
prompt-toolkit @ file:///tmp/build/80754af9/prompt-toolkit_1598885458782/work
pycparser @ file:///tmp/build/80754af9/pycparser_1594388511720/work
Pygments @ file:///tmp/build/80754af9/pygments_1600458456400/work
pyparsing==2.4.7
pyrsistent @ file:///C:/ci/pyrsistent_1600141795814/work
python-dateutil==2.8.1
pywin32==227
pywinpty==0.5.7
pyzmq==19.0.2
qtconsole @ file:///tmp/build/80754af9/qtconsole_1600870028330/work
QtPy==1.9.0
Send2Trash==1.5.0
sip==4.19.13
six==1.15.0
terminado==0.8.3
testpath==0.4.4
tornado==6.0.4
traitlets==4.3.3
wcwidth @ file:///tmp/build/80754af9/wcwidth_1593447189090/work
webencodings==0.5.1
widgetsnbextension==3.5.1
wincertstore==0.2
zipp==3.1.0
```

4. What did you do?
If possible, provide a recipe for reproducing the error.
A complete runnable program is good.

pip install snowflake-connector-python[pandas]

5. What did you expect to see?
Successful installation

6. What did you see instead?
ERROR: Failed building wheel for snowflake-connector-python
Failed to build snowflake-connector-python
ERROR: Could not build wheels for snowflake-connector-python which use PEP 517 and cannot be installed directly

7. Can you set logging to DEBUG and collect the logs?

**Comment by sfc-gh-mkeller** [comment]
We currently don't currently support Windows with Python 3.8.
Please use Python 3.7 for now and sorry about the inconvenience!

**Comment by zbrookle** [comment]
@sfc-gh-mkeller Is there a reason for not supporting python 3.8 on windows?

**Comment by sfc-gh-mkeller** [comment]
@zbrookle yes, let me see if I could find our previous discussion about this: https://github.com/snowflakedb/snowflake-connector-python/issues/295
We just didn't have the time to revisit this issue since then.
Explanation to what changed in Python 3.8 is here: https://docs.python.org/3/whatsnew/3.8.html#os but when we tried we couldn't get it working.
Would appreciate some help from folks that are knowledgeable on Windows and Python 3.8

**Comment by info-rchitect** [comment]
@sfc-gh-mkeller Any update here?

**Comment by sfc-gh-mkeller** [comment]
@info-rchitect This has been fixed with #527
This was a pretty interesting issue, feel free to read the comments on that PR 😄

**Comment by info-rchitect** [comment]
@sfc-gh-mkeller  Yeah lazy instantiation is a nice solution.  Very tough to debug I bet!  I am using the python snowflake API to asynchronously submit extractions based on a UI in Jupyter Notebook/Lab so this fix is very much appreciated.
